src/controllers/post.controller.js

```diff
@@ -24,58 +25,65 @@ const addNewPost = asyncHandler(async (req, res, next) => {
   res.status(201).json(new ApiResponse(201, { newPost }, "Post created"));
 });
 
-// ------------------------- Update Post -------------------------
+// ------------------------- Update self Post -------------------------
 const updatePost = asyncHandler(async (req, res, next) => {
-  const { postId, image, caption, likes, comments } = req.body;
+  const { postId, caption } = req.body;
+  const userIdFromToken = req.user._id;
 
-  if (!(caption && postId)) {
+  if (bodyDataExists(postId && caption)) {
     throw new ApiError(400, "Incomplete Data");
   }
-  if (comments) {
-    comments.forEach((comment) => {
-      if (!(comment.userId && comment.comment)) {
-        throw new ApiError(400, "Incomplete Data");
-      }
-    });
+
+  const existingPost = await Post.findOne({ _id: postId });
+  if (!existingPost) {
+    throw new ApiError(500, "Post not found");
   }
-  if (likes) {
-    if (likes.length === 0) throw new ApiError(400, "Incomplete Data");
+
+  if (existingPost.userId.toString() != userIdFromToken) {
+    throw new ApiError(403, "Unauthorized");
   }
 
-  const existingPost = await Post.findOneAndUpdate(
+  const updatedPost = await Post.findOneAndUpdate(
```

Comment:
You already fetched the post, just update that instead of doing one more DB call

Reply:
ok.